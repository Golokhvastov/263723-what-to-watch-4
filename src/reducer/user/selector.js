import NameSpace from "../name-space.js";

const NAME_SPACE = NameSpace.USER;

const SERVER_URL = `https://4.react.pages.academy`;

export const getAuthorizationStatus = (state) => {
  return state[NAME_SPACE].authorizationStatus;
};

const adaptUserData = (userData) => {
  if (userData) {
    if (userData.avatar_url) {
      return {
        id: userData.id,
        email: userData.email,
        name: userData.name,
        avatarUrl: SERVER_URL + userData.avatar_url,
      };
    }
  }

  return userData;
};

export const getUserInfo = (state) => {
  return adaptUserData(state[NAME_SPACE].userInfo);
};
