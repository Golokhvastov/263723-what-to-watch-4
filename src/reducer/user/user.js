import {extend} from "../../utils/utils.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
};

const ActionCreator = {
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status
  }),
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData, onSuccess, onError) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then(() => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        onSuccess();
      })
      .catch((err) => {
        onError();
        throw err;
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload
      });
  }

  return state;
};

export {reducer, ActionType, ActionCreator, AuthorizationStatus, Operation};
