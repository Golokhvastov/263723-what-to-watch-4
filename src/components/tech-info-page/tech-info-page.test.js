import React from "react";
import renderer from "react-test-renderer";
import TechInfoPage from "./tech-info-page.jsx";
import {AuthorizationStatus} from "../../reducer/user/user.js";

const mocks = {
  userInfo: {
    avatarUrl: `test avatarUrl.jpg`,
  },
};

it(`TechInfoPage render correctly`, () => {
  const tree = renderer
    .create(
        <TechInfoPage
          authorizationStatus = {AuthorizationStatus.AUTH}
          userInfo = {mocks.userInfo}
          onLogoClick = {() => {}}
          onSignInClick = {() => {}}
          onAvatarClick = {() => {}}
        >
          <div>Test1</div>
        </TechInfoPage>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
