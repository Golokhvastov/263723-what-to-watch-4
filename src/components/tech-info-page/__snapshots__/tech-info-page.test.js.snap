// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechInfoPage render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    />
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="main.html"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <a
            className="user-block__link"
            href="login.html"
            onClick={[Function]}
          >
            <img
              alt="User avatar"
              height="63"
              src="test avatarUrl.jpg"
              width="63"
            />
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        />
        <div>
          Test1
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
      </ul>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
